neqo-transport/src/cc/classic_cc.rs

```diff
@@ -382,16 +401,21 @@ impl<T: WindowAdjustment> ClassicCongestionControl<T> {
 
     #[allow(clippy::unused_self)]
     fn app_limited(&self) -> bool {
-        //TODO(agrover): how do we get this info??
-        false
+        if self.bytes_in_flight >= self.congestion_window {
+            false
+        } else if self.state.is_slow_start() {
+            self.bytes_in_flight < self.congestion_window / 2
+        } else {
+            self.congestion_window > (self.bytes_in_flight + MAX_DATAGRAM_SIZE * PACING_BURST_SIZE)
```

Comment:
Symmetry is important here.  This always puts the in-flight value first.

```suggestion
            // Allow for potential doubling of the congestion window during slow start.
            // That is, the application might not have been able to send enough to respond
            // to increases to the congestion window.
            self.bytes_in_flight < self.congestion_window / 2
        } else {
            // We're not limited if the in-flight data is within a single burst of the congestion window.
            (self.bytes_in_flight + MAX_DATAGRAM_SIZE * PACING_BURST_SIZE) < self.congestion_window
```